bgpd/bgp_route.c

```diff
@@ -11351,8 +11353,18 @@ static int bgp_table_stats_walker(struct thread *t)
 	return 0;
 }
 
-static int bgp_table_stats(struct vty *vty, struct bgp *bgp, afi_t afi,
-			   safi_t safi, struct json_object *json_array)
+static void bgp_table_stats_all(struct vty *vty, afi_t afi, safi_t safi,
+				struct json_object *json_array)
+{
+	struct listnode *node, *nnode;
+	struct bgp *bgp;
+
+	for (ALL_LIST_ELEMENTS(bm->bgp, node, nnode, bgp))
```

Comment:
In this below CLI. if BGP is null then we do early return. BGP can be null when vrf is set to "ALL". I guess that check can be removed as you have taken care in your function.

DEFUN(show_ip_bgp_statistics_all, show_ip_bgp_statistics_all_cmd,
      "show [ip] bgp [<view|vrf> VIEWVRFNAME] statistics-all [json]",
      SHOW_STR IP_STR BGP_STR BGP_INSTANCE_HELP_STR
      "Display number of prefixes for all afi/safi\n" JSON_STR)
{
	bool uj = use_json(argc, argv);
	struct bgp *bgp = NULL;
	safi_t safi = SAFI_UNICAST;
	afi_t afi = AFI_IP6;
	int idx = 0;
	struct json_object *json_all = NULL;
	struct json_object *json_afi_safi = NULL;

	bgp_vty_find_and_parse_afi_safi_bgp(vty, argv, argc, &idx, &afi, &safi,
					    &bgp, false);
	if (!bgp)
		return CMD_WARNING;

Reply:
@Spantik you are talking about a different command `show_ip_bgp_statistics_all_cmd`.

Reply:
This should be fixed as well for `show_ip_bgp_statistics_all_cmd`. Maybe with a separate PR.

Reply:
but this command uses the same function.